examples/src/bin/wifi_embassy_access_point.rs

```diff
@@ -197,6 +210,46 @@ async fn main(spawner: Spawner) -> ! {
     }
 }
 
+#[embassy_executor::task]
+async fn run_dhcp(
+    stack: &'static Stack<WifiDevice<'static, WifiApDevice>>,
+    gw_ip_addr: &'static str,
+) {
+    use core::net::{Ipv4Addr, SocketAddrV4};
+
+    use edge_dhcp::{
+        io::{self, DEFAULT_SERVER_PORT},
+        server::{Server, ServerOptions},
+    };
+    use edge_nal::UdpBind;
+    use edge_nal_embassy::{Udp, UdpBuffers};
+
+    let ip = Ipv4Addr::from_str(gw_ip_addr).expect("dhcp task failed to parse gw ip");
+
+    let mut buf = [0u8; 1500];
+
+    let mut gw_buf = [Ipv4Addr::UNSPECIFIED];
+
+    let buffers = UdpBuffers::<3, 1024, 1024, 10>::new();
+    let unbound_socket = Udp::new(stack, &buffers);
+    let mut bound_socket = unbound_socket
+        .bind(core::net::SocketAddr::V4(SocketAddrV4::new(
+            Ipv4Addr::UNSPECIFIED,
+            DEFAULT_SERVER_PORT,
+        )))
+        .await
+        .unwrap();
+
+    _ = io::server::run(
```

Comment:
This should probably be in a loop. If the server returns an error it won't restart again.

Reply:
Should I put a delay between server restarts?

Reply:
I don't think it makes much of a difference, having a delay is fine though.